plugin/trino-hive/src/main/java/io/trino/plugin/hive/DirectoryCallback.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import io.trino.plugin.hive.metastore.Partition;
+import io.trino.plugin.hive.metastore.Table;
+import org.apache.hadoop.fs.Path;
+
+public interface DirectoryCallback
+{
+    DirectoryCallback NOOP = new DirectoryCallback() {
+        @Override
+        public void onTableUpdate(Table table, Path path)
+        {
+        }
+
+        @Override
+        public void onPartitionUpdate(Partition partition, Path path)
+        {
+        }
+
+        @Override
+        public void onTableDropped(Table table, Path path)
+        {
+        }
+
+        @Override
+        public void onPartitionDropped(Partition partition, Path path)
+        {
+        }
+
+        @Override
+        public void onTableAltered(Table table, Path path)
+        {
+        }
+
+        @Override
+        public void onPartitionAltered(Partition partition, Path path)
+        {
+        }
+    };
+
+    void onTableUpdate(Table table, Path path);
+
+    void onPartitionUpdate(Partition partition, Path path);
```

Comment:
The API is modeled around tables and partitions.
if we keep this, rename the class to `TableChangeCallback`